Lib/test/test_strptime.py

```diff
@@ -312,6 +365,17 @@ def test_year(self):
                                 "'y' test failed; passed in '%s' "
                                 "and returned '%s'" % (bound, strp_output[0]))
 
+    def test_bad_year(self):
+        for directive, bad_inputs in (
+            ('y', ('9', '100', 'ni')),
+            ('Y', ('7', '42', '999', '10000', 'SPAM')),
+        ):
+            fmt = "%" + directive
```

Comment:
Maybe you can use f-string.

Reply:
Both ways are valid; the one I chose is more consistent with the existing code in this module.